website/src/content/blog/2025-03-31-typespec-1-0-release/type_spec_1_0_release.md

```diff
@@ -0,0 +1,202 @@
+}
+```
+
+With the generated code in place, you can quickly integrate it into your project, build the service, and run it. TypeSpec ensures that your API implementation stays perfectly aligned with your specification, saving you time and effort.
+
+- **Note:** The C# service codegen is currently in preview, so there may be gaps in the implementation. We want service codegen to be rock solid when we move from RC to full release, so we encourage you to try it out and tell us if/when it breaks to help us improve this feature.
+
+### Client-side code generation
+
+In addition to server-side code, TypeSpec can generate client-side libraries for multiple languages, including JavaScript, Python, C#, and Java, with support for additional languages like Rust in development.
+
+For example, a generated JavaScript library will allow you to interact with your `todo` API like this:
+
+```javascript
+const client = new TodoServiceClient();
+const todo = await client.createTodoItem({
+  content: "Buy groceries",
+  dueDate: "2025-03-21",
+  isCompleted: false,
+});
+console.log(todo);
+```
+
+TypeSpec simplifies maintenance by allowing seamless updates to server and client code as APIs evolve, enabling you to focus on business logic rather than repetitive tasks.
+
+## Emitters: From TypeSpec to the API format you need
+
+TypeSpec's core value proposition is the ability to define your API once and generate multiple output formats. The 1.0-RC release ships with a stable OpenAPI 3.0 emitter (`@typespec/openapi3`), enabling you to seamlessly integrate TypeSpec into existing OpenAPI workflows and toolchains.
+
+But our emitter story doesn't stop there. The TypeSpec ecosystem includes several additional emitters:
+
+- **OpenAPI 2.0 Emitter**: For teams that still require Swagger/OpenAPI 2.0 compatibility
+- **REST Emitter**: Produces JSON descriptions of REST APIs
+- **Protocol Buffer Emitter**: Generates Protocol Buffer (protobuf) definitions
+- **JsonSchema Emitter**: Creates JSON Schema documents from your TypeSpec types
+
+While some of these emitters remain in preview as we gather feedback, they demonstrate the extensibility of TypeSpec across different API description formats.
+
+For teams with existing OpenAPI definitions looking to adopt TypeSpec, we've also created the OpenAPI to TypeSpec converter tool. This tool helps you migrate your existing API definitions to TypeSpec, providing a smoother onboarding experience and preserving your investment in OpenAPI. You can find the converter in our [GitHub repository](https://github.com/microsoft/typespec/tree/main/packages/openapi3-to-typespec).
+
+For teams with specialized needs, TypeSpec 1.0-RC includes a new [emitter framework](https://github.com/alloy-framework/alloy) that allows you to build custom emitters. This framework provides access to TypeSpec's compiler APIs and type system, enabling you to transform your TypeSpec definitions into any format your organization requires.
```

Comment:
Alloy isn't the emitter framework but the base of the emitter framework. Suggestion:

Of course, TypeSpec couldn't deliver on its mission to be the source of truth for myriad API formats without making it easy to add new ones. We've started work on an emitter framework that we believe will provide the foundation for a robust ecosystem of emitters and reusable components for generating language syntax, schema languages, REST API patterns, and so forth. It should be familiar to JavaScript developers with experience in React, but is approachable by developers of any experience level.

Reply:
I hope to finish the EF docs by tomorrow and we can link to it here.